--- origsrc/vlc-0.8.6i/modules/codec/realaudio.c	2008-07-08 15:59:23.000000000 -0500
+++ src/vlc-0.8.6i/modules/codec/realaudio.c	2008-07-17 15:21:06.437500000 -0500
@@ -460,6 +460,9 @@
 
 static int OpenWin32Dll( decoder_t *p_dec, char *psz_path, char *psz_dll )
 {
+#if defined(__CYGWIN__)
+    return OpenNativeDll( p_dec, psz_path, psz_dll );
+#endif
 #if defined(LOADER) || defined(WIN32)
     decoder_sys_t *p_sys = p_dec->p_sys;
     void *handle = 0, *context = 0;
