--- origsrc/vlc-0.8.6i/modules/misc/network/ipv4.c	2008-07-08 15:59:24.000000000 -0500
+++ src/vlc-0.8.6i/modules/misc/network/ipv4.c	2008-07-17 15:21:06.484375000 -0500
@@ -265,7 +265,7 @@
         if (( rem.sin_addr.s_addr != INADDR_ANY )
          /*&& ((ntohl (loc.sin_addr.s_addr) >> 24) == 232)*/)
         {
-#ifndef IP_ADD_SOURCE_MEMBERSHIP
+#if !defined( IP_ADD_SOURCE_MEMBERSHIP ) || defined ( __CYGWIN__ )
             errno = ENOSYS;
 #else
             struct ip_mreq_source imr;
--- origsrc/vlc-0.8.6i/src/stream_output/sap.c	2008-07-08 16:08:36.000000000 -0500
+++ src/vlc-0.8.6i/src/stream_output/sap.c	2008-07-17 15:21:06.531250000 -0500
@@ -279,7 +279,7 @@
 
     switch( addr.ss_family )
     {
-#if defined (HAVE_INET_PTON) || defined (WIN32)
+#if defined (AF_INET6) || defined (WIN32)
         case AF_INET6:
         {
             /* See RFC3513 for list of valid IPv6 scopes */
@@ -456,7 +456,7 @@
     psz_head[2] = (i_hash & 0xFF00) >> 8; /* Msg id hash */
     psz_head[3] = (i_hash & 0xFF);        /* Msg id hash 2 */
 
-#if defined (HAVE_INET_PTON) || defined (WIN32)
+#if defined (AF_INET6) || defined (WIN32)
     if( b_ipv6 )
     {
         inet_pton( AF_INET6, /* can't fail */
