--- origsrc/vlc-0.9.8a/configure.ac	2009-02-08 23:07:00.623125000 -0600
+++ src/vlc-0.9.8a/configure.ac	2009-02-08 23:17:34.716875000 -0600
@@ -4016,9 +4016,9 @@ dnl  Check for the Xinerama extension
 dnl
 AC_ARG_ENABLE(xinerama,
   [  --enable-xinerama       Xinerama support (default enabled)])
-if test "${enable_xvideo}" != "no" && test "${enable_xinerama}" != "no" &&
+if test "${enable_xinerama}" != "no" &&
   (test "${SYS}" != "mingw32" -a "${SYS}" != "mingwce" ||
-   test "${enable_xvideo}" = "yes"); then
+   test "${enable_xinerama}" = "yes"); then
   ac_cv_have_xinerama="no"
   CPPFLAGS="${CPPFLAGS_save} ${X_CFLAGS}"
   CFLAGS="${CFLAGS_save} ${X_LIBS} ${X_PRE_LIBS} -lX11 -lXext"
