--- origsrc/vlc-0.9.0-test2/modules/misc/freetype.c	2008-07-05 06:34:50.000000000 -0500
+++ src/vlc-0.9.0-test2/modules/misc/freetype.c	2008-07-17 19:49:01.296875000 -0500
@@ -60,6 +60,9 @@
 #elif defined( WIN32 )
 #define DEFAULT_FONT "" /* Default font found at run-time */
 #define FC_DEFAULT_FONT "Arial"
+#elif defined( __CYGWIN__ )
+#define DEFAULT_FONT "/usr/share/fonts/TTF/DejaVuSerif-Bold.ttf"
+#define FC_DEFAULT_FONT "Serif Bold"
 #else
 #define DEFAULT_FONT "/usr/share/fonts/truetype/freefont/FreeSerifBold.ttf"
 #define FC_DEFAULT_FONT "Serif Bold"
