--- origsrc/vlc-0.9.0-test2/modules/access/mmap.c	2008-07-05 06:34:47.000000000 -0500
+++ src/vlc-0.9.0-test2/modules/access/mmap.c	2008-07-17 23:19:43.968750000 -0500
@@ -129,7 +129,7 @@
             goto error;
     }
 
-    p_sys->page_size = sysconf (_SC_PAGE_SIZE);
+    p_sys->page_size = sysconf (_SC_PAGESIZE);
     p_sys->mtu = MMAP_SIZE;
     if (p_sys->mtu < p_sys->page_size)
         p_sys->mtu = p_sys->page_size;
