--- origsrc/vlc-0.9.0-test2/src/misc/threads.c	2008-07-11 16:01:53.000000000 -0500
+++ src/vlc-0.9.0-test2/src/misc/threads.c	2008-07-17 22:36:02.265625000 -0500
@@ -104,7 +104,7 @@
 
     /* Sometimes strerror_r() crashes too, so make sure we print an error
      * message before we invoke it */
-#ifdef __GLIBC__
+#if defined(__GLIBC__) || defined(__CYGWIN__)
     /* Avoid the strerror_r() prototype brain damage in glibc */
     errno = error;
     fprintf (stderr, " Error message: %m at:\n");
