--- origsrc/vlc-1.1.7/modules/demux/mkv/matroska_segment_parse.cpp	2011-01-30 13:39:25.000000000 -0600
+++ src/vlc-1.1.7/modules/demux/mkv/matroska_segment_parse.cpp	2011-02-01 13:55:51.125036700 -0600
@@ -22,6 +22,8 @@
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston MA 02110-1301, USA.
  *****************************************************************************/
 
+#include <time.h>
+
 #include "matroska_segment.hpp"
 
 #include "chapters.hpp"
--- origsrc/vlc-1.1.7/src/input/item.c	2010-11-02 17:23:18.000000000 -0500
+++ src/vlc-1.1.7/src/input/item.c	2011-02-01 07:41:47.069085300 -0600
@@ -25,6 +25,7 @@
 # include "config.h"
 #endif
 #include <assert.h>
+#include <time.h>
 
 #include <vlc_common.h>
 #include <vlc_url.h>
--- origsrc/vlc-1.1.7/src/video_output/video_epg.c	2010-04-12 19:22:28.000000000 -0500
+++ src/vlc-1.1.7/src/video_output/video_epg.c	2011-02-01 07:42:58.470409800 -0600
@@ -23,6 +23,7 @@
 #ifdef HAVE_CONFIG_H
 # include "config.h"
 #endif
+#include <time.h>
 
 #include <vlc_common.h>
 #include <vlc_vout.h>
