--- origsrc/vlc-1.1.7/modules/misc/freetype.c	2010-11-13 13:34:58.000000000 -0600
+++ src/vlc-1.1.7/modules/misc/freetype.c	2011-02-01 05:32:02.069943900 -0600
@@ -66,6 +66,9 @@
 #elif defined( HAVE_MAEMO )
 #define DEFAULT_FONT "/usr/share/fonts/nokia/nosnb.ttf"
 #define FC_DEFAULT_FONT "Nokia Sans Bold"
+#elif defined( __CYGWIN__ )
+#define DEFAULT_FONT "/usr/share/fonts/TTF/DejaVuSerif-Bold.ttf"
+#define FC_DEFAULT_FONT "Serif Bold"
 #else
 #define DEFAULT_FONT "/usr/share/fonts/truetype/freefont/FreeSerifBold.ttf"
 #define FC_DEFAULT_FONT "Serif Bold"
