--- origsrc/vlc-1.1.7/configure.ac	2011-01-30 13:33:58.000000000 -0600
+++ src/vlc-1.1.7/configure.ac	2011-02-01 07:18:23.963659900 -0600
@@ -302,7 +302,7 @@ case "${host_os}" in
         export MACOSX_DEPLOYMENT_TARGET
     fi
     ;;
-  *mingw32* | *cygwin* | *wince* | *mingwce*)
+  *mingw32* | *wince* | *mingwce*)
     AC_CHECK_TOOL(WINDRES, windres, :)
     AC_CHECK_TOOL(OBJCOPY, objcopy, :)
 
@@ -946,7 +946,7 @@ dnl Check for dbus
 AC_ARG_ENABLE(dbus,
   [  --enable-dbus           Linux D-BUS message bus system (default enabled)])
 case "${SYS}" in
-    linux*|*bsd*)
+    linux*|*bsd*|cygwin*)
 if test "${enable_dbus}" != "no" -a "${SYS}" != "mingw32" -a "${SYS}" != "mingwce"
 then
   dnl api stable dbus
@@ -975,10 +975,12 @@ then
           VLC_ADD_LIBS([telepathy],[$DBUS_LIBS])
           VLC_ADD_CFLAGS([telepathy],[$DBUS_CFLAGS])
         fi
+        if test "${SYS}" != "cygwin"; then
         dnl Power Management Inhibiter
         VLC_ADD_PLUGIN([inhibit])
         VLC_ADD_LIBS([inhibit],[$DBUS_LIBS])
         VLC_ADD_CFLAGS([inhibit],[$DBUS_CFLAGS])
+        fi
     ],
     [AC_MSG_ERROR([Couldn't find DBus >= 1.0.0, install libdbus-dev ?])]
   )
@@ -2892,7 +2892,7 @@ dnl
 AC_ARG_ENABLE(quicktime,
   [  --enable-quicktime      QuickTime module (deprecated)])
 if test "${enable_quicktime}" = "yes"; then
-  if test "${SYS}" = "mingw32"; then
+  if test "${SYS}" = "mingw32" -o "${SYS}" = "cygwin"; then
     VLC_ADD_PLUGIN([quicktime])
   else
   AC_CHECK_HEADERS(QuickTime/QuickTime.h,
@@ -3317,7 +3319,10 @@ AS_IF([test "${enable_xcb}" != "no"], [
   dnl libxcb
   PKG_CHECK_MODULES(XCB, [xcb])
   PKG_CHECK_MODULES(XCB_SHM, [xcb-shm])
-  VLC_ADD_PLUGIN([xscreensaver xcb_x11 xcb_window xcb_screen xcb_apps])
+  VLC_ADD_PLUGIN([xcb_x11 xcb_window xcb_screen xcb_apps])
+  if test "${SYS}" != "cygwin"; then
+    VLC_ADD_PLUGIN([xscreensaver])
+  fi
 
   AS_IF([test "${enable_xvideo}" != "no"], [
     PKG_CHECK_MODULES(XCB_XV, [xcb-xv >= 1.1.90.1], [
@@ -3357,7 +3362,9 @@ AS_IF([test "${enable_xcb}" != "no"], [
     VLC_ADD_LIBS([globalhotkeys], [${XCB_KEYSYMS_LIBS} ${XCB_LIBS}])
     VLC_ADD_CFLAGS([xcb_window], [-DHAVE_XCB_KEYSYMS])
   ])
+  if test "${SYS}" != "cygwin"; then
   VLC_ADD_PLUGIN([xdg_screensaver])
+  fi
 ])
 
 AC_ARG_ENABLE(glx,
