--- origsrc/vlc-1.1.10/modules/misc/freetype.c
+++ src/vlc-1.1.10/modules/misc/freetype.c
@@ -66,6 +66,9 @@
 #elif defined( HAVE_MAEMO )
 # define DEFAULT_FONT_FILE "/usr/share/fonts/nokia/nosnb.ttf"
 # define DEFAULT_FAMILY "Nokia Sans Bold"
+#elif defined( __CYGWIN__ )
+# define DEFAULT_FONT_FILE "/usr/share/fonts/TTF/DejaVuSerif-Bold.ttf"
+# define DEFAULT_FAMILY "Serif Bold"
 #else
 # define DEFAULT_FONT_FILE "/usr/share/fonts/truetype/freefont/FreeSerifBold.ttf"
 # define DEFAULT_FAMILY "Serif Bold"
