--- origsrc/vlc-1.1.7/configure.ac	2011-01-30 13:33:58.000000000 -0600
+++ src/vlc-1.1.7/configure.ac	2011-02-01 07:18:23.963659900 -0600
@@ -302,7 +302,7 @@ case "${host_os}" in
         export MACOSX_DEPLOYMENT_TARGET
     fi
     ;;
-  *mingw32* | *cygwin* | *wince* | *mingwce*)
+  *mingw32* | *wince* | *mingwce*)
     AC_CHECK_TOOL(WINDRES, windres, :)
     AC_CHECK_TOOL(OBJCOPY, objcopy, :)
 
@@ -946,7 +946,7 @@ dnl Check for dbus
 AC_ARG_ENABLE(dbus,
   [  --enable-dbus           Linux D-BUS message bus system (default enabled)])
 case "${SYS}" in
-    linux*|*bsd*)
+    linux*|*bsd*|cygwin*)
 if test "${enable_dbus}" != "no" -a "${SYS}" != "mingw32" -a "${SYS}" != "mingwce"
 then
   dnl api stable dbus
@@ -2892,7 +2892,7 @@ dnl
 AC_ARG_ENABLE(quicktime,
   [  --enable-quicktime      QuickTime module (deprecated)])
 if test "${enable_quicktime}" = "yes"; then
-  if test "${SYS}" = "mingw32"; then
+  if test "${SYS}" = "mingw32" -o "${SYS}" = "cygwin"; then
     VLC_ADD_PLUGIN([quicktime])
   else
   AC_CHECK_HEADERS(QuickTime/QuickTime.h,
