--- origsrc/vlc-1.1.7/modules/misc/inhibit/xdg.c	2010-11-26 16:32:46.000000000 -0600
+++ src/vlc-1.1.7/modules/misc/inhibit/xdg.c	2011-02-01 15:25:26.982455700 -0600
@@ -26,7 +26,11 @@
 #include <vlc_plugin.h>
 #include <vlc_inhibit.h>
 #include <assert.h>
+#ifdef __CYGWIN__
+#include <process.h>
+#else
 #include <spawn.h>
+#endif
 #include <sys/wait.h>
 
 static int Open (vlc_object_t *);
@@ -137,7 +141,12 @@ static void *Thread (void *data)
         pid_t pid;
 
         vlc_mutex_unlock (&p_sys->lock);
+#ifdef __CYGWIN__
+        pid = execvpe ("xdg-screensaver", argv, environ);
+        if (pid != -1)
+#else
         if (!posix_spawnp (&pid, "xdg-screensaver", NULL, NULL, argv, environ))
+#endif
         {
             int status;
 
--- origsrc/vlc-1.1.7/modules/misc/inhibit/xscreensaver.c	2010-04-12 19:22:27.000000000 -0500
+++ src/vlc-1.1.7/modules/misc/inhibit/xscreensaver.c	2011-02-01 15:24:45.486383400 -0600
@@ -39,7 +39,11 @@
 #include <sys/wait.h>
 #include <fcntl.h>
 #include <signal.h>
+#ifdef __CYGWIN__
+#include <process.h>
+#else
 #include <spawn.h>
+#endif
 
 /*****************************************************************************
  * Local prototypes
@@ -53,8 +57,10 @@ static void Inhibit( vlc_inhibit_t *, bo
 struct vlc_inhibit_sys
 {
     vlc_timer_t timer;
+#ifndef __CYGWIN__
     posix_spawn_file_actions_t actions;
     posix_spawnattr_t attr;
+#endif
     int nullfd;
 };
 
@@ -89,6 +95,7 @@ static int Activate( vlc_object_t *p_thi
     p_ih->inhibit = Inhibit;
 
     int fd = vlc_open ("/dev/null", O_WRONLY);
+#ifndef __CYGWIN__
     posix_spawn_file_actions_init (&p_sys->actions);
     if (fd != -1)
     {
@@ -102,6 +109,7 @@ static int Activate( vlc_object_t *p_thi
     posix_spawnattr_init (&p_sys->attr);
     sigemptyset (&set);
     posix_spawnattr_setsigmask (&p_sys->attr, &set);
+#endif
    
     return VLC_SUCCESS;
 }
@@ -117,8 +125,10 @@ static void Deactivate( vlc_object_t *p_
     vlc_timer_destroy( p_sys->timer );
     if (p_sys->nullfd != -1)
         close (p_sys->nullfd);
+#ifndef __CYGWIN__
     posix_spawnattr_destroy (&p_sys->attr);
     posix_spawn_file_actions_destroy (&p_sys->actions);
+#endif
     free( p_sys );
 }
 
@@ -136,8 +146,13 @@ static void Execute (vlc_inhibit_t *p_ih
     vlc_inhibit_sys_t *p_sys = p_ih->p_sys;
     pid_t pid;
 
+#ifdef __CYGWIN__
+    pid = execvpe (argv[0], argv, environ);
+    if (pid != -1)
+#else
     if (posix_spawnp (&pid, argv[0], &p_sys->actions, &p_sys->attr,
                       (char **)argv, environ) == 0)
+#endif
     {
         while (waitpid (pid, NULL, 0) != pid);
     }
