--- origsrc/vlc-2.0.2/modules/access/bluray.c	2012-06-03 16:39:15.000000000 -0500
+++ src/vlc-2.0.2/modules/access/bluray.c	2012-07-02 21:49:51.880554800 -0500
@@ -186,7 +186,7 @@ static int blurayOpen( vlc_object_t *obj
             /* bd_path may be a symlink (e.g. /dev/dvd -> /dev/sr0), so make
              * sure we look up the real device */
             char* bd_device = realpath(bd_path, NULL);
-            while ((m = getmntent_r (mtab, &mbuf, buf, sizeof(buf))) != NULL) {
+            while ((m = getmntent (mtab)) != NULL) {
                 if (!strcmp (m->mnt_fsname, (bd_device == NULL ? bd_path : bd_device))) {
                     strncpy (bd_path, m->mnt_dir, sizeof(bd_path));
                     bd_path[sizeof(bd_path) - 1] = '\0';
