--- origsrc/vlc-2.1.5/modules/misc/Modules.am	2013-12-03 03:12:34.000000000 -0600
+++ src/vlc-2.1.5/modules/misc/Modules.am	2015-01-15 22:35:59.723996000 -0600
@@ -34,7 +34,7 @@ libxdg_screensaver_plugin_la_SOURCES = i
 libxdg_screensaver_plugin_la_CFLAGS = $(AM_CFLAGS)
 libxdg_screensaver_plugin_la_LIBADD = $(AM_LIBADD)
 if HAVE_XCB
-libvlc_LTLIBRARIES += \
+EXTRA_LTLIBRARIES += \
 	libxdg_screensaver_plugin.la
 endif
 
