--- origsrc/vlc-2.1.5/modules/text_renderer/freetype.c	2014-02-14 11:40:50.000000000 -0600
+++ src/vlc-2.1.5/modules/text_renderer/freetype.c	2015-01-15 22:16:37.353512200 -0600
@@ -67,9 +67,9 @@
 # define SYSTEM_DEFAULT_MONOSPACE_FONT_FILE "/system/fonts/DroidSansMono.ttf"
 # define SYSTEM_DEFAULT_MONOSPACE_FAMILY "Droid Sans Mono"
 #else
-# define SYSTEM_DEFAULT_FONT_FILE "/usr/share/fonts/truetype/freefont/FreeSerifBold.ttf"
+# define SYSTEM_DEFAULT_FONT_FILE "/usr/share/fonts/TTF/FreeSerifBold.ttf"
 # define SYSTEM_DEFAULT_FAMILY "Serif Bold"
-# define SYSTEM_DEFAULT_MONOSPACE_FONT_FILE "/usr/share/fonts/truetype/freefont/FreeMono.ttf"
+# define SYSTEM_DEFAULT_MONOSPACE_FONT_FILE "/usr/share/fonts/TTF/FreeMono.ttf"
 # define SYSTEM_DEFAULT_MONOSPACE_FAMILY "Monospace"
 #endif
 
