--- origsrc/vlc-0.8.6i/bindings/java/Makefile.am	2008-07-08 15:59:22.000000000 -0500
+++ src/vlc-0.8.6i/bindings/java/Makefile.am	2008-07-17 15:21:06.343750000 -0500
@@ -53,8 +53,12 @@
 LIBJINCLUDES = -L$(JAVA_HOME)/jre/lib/$(PROCESSOR_FAMILY) -ljawt
 JCH = gcjh -jni
 JAVACXXFLAGS = `top_builddir=../.. ../../vlc-config --cflags pic` -I../../ -I ../../include $(JINCLUDES)
+if USE_LIBTOOL
+JAVALDFLAGS = -L../../src/.libs -lvlc `top_builddir=../.. ../../vlc-config --libs vlc pic`
+else
 JAVALDFLAGS = -L../../src -lvlc `top_builddir=../.. ../../vlc-config --libs builtin vlc pic`
 endif
+endif
 
 export JINCLUDES
 
@@ -68,14 +72,14 @@
 else
 libjvlc.so: $(OBJECTS) 	
 	$(MAKE) -C src
-	$(CXX) -shared $(COBJECTS) $(LIBJINCLUDES) $(LDFLAGS) $(JAVALDFLAGS) -o libjvlc.so
+	$(CXX) -shared $(COBJECTS) $(LIBJINCLUDES) $(LDFLAGS) $(JAVALDFLAGS) -o cygjvlc.dll
 endif
 
 VlcClient: $(OBJECTS)
-	$(JAVAC) VlcClient.java
+	$(JAVAC) -d . $(srcdir)/VlcClient.java
 
 VLCExample: $(OBJECTS)
-	$(JAVAC) VLCExample.java
+	$(JAVAC) -d . $(srcdir)/VLCExample.java
 
 if HAVE_WIN32
 %.class: %.java 
@@ -83,7 +87,7 @@
 	$(JCH) org.videolan.jvlc.$(*F)
 else
 %.class: %.java
-	$(JAVAC) $?
+	$(JAVAC) -d . -classpath $(srcdir) $?
 	$(JCH) $(@D)/$(*F) -o includes/$(*F).h
 endif
 
--- origsrc/vlc-0.8.6i/bindings/java/src/Makefile.am	2008-07-08 15:59:22.000000000 -0500
+++ src/vlc-0.8.6i/bindings/java/src/Makefile.am	2008-07-17 15:21:06.359375000 -0500
@@ -13,7 +13,7 @@
 
 if BUILD_JAVA
 
-JAVACXXFLAGS = -I. -Isrc -I../../ -I ../../include -I../../../include  $(JINCLUDES) `top_builddir=../../.. ../../../vlc-config --cflags pic` -c
+JAVACXXFLAGS = -I. -Isrc -I../../ -I ../../include -I$(top_srcdir)/include  $(JINCLUDES) `top_builddir=../../.. ../../../vlc-config --cflags pic` -c
 
 
 all: $(COBJECTS)
--- origsrc/vlc-0.8.6i/bindings/mediacontrol-python/Makefile.am	2008-07-08 15:59:22.000000000 -0500
+++ src/vlc-0.8.6i/bindings/mediacontrol-python/Makefile.am	2008-07-17 15:21:06.375000000 -0500
@@ -17,8 +17,8 @@
 
 # FIXME: python setup.py install does not have any option to install from a different build directory
 # so this will not work in a separate builddir
-install:
-	python $(srcdir)/setup.py install
+#install:
+#	python $(srcdir)/setup.py install
 
 clean:
 	$(RM) -rf build
--- origsrc/vlc-0.8.6i/bindings/mediacontrol-python/setup.py	2008-07-08 15:59:22.000000000 -0500
+++ src/vlc-0.8.6i/bindings/mediacontrol-python/setup.py	2008-07-17 15:21:06.390625000 -0500
@@ -25,7 +25,7 @@
     # Do not use PIC version on win32 and Mac OS X
 if True:
     # PIC version seems to be disabled on all platforms
-    vlclib=os.path.join( top_builddir, 'src', 'libvlc.a' )
+    vlclib=os.path.join( top_builddir, 'src', '.libs', 'libvlc.dll.a' )
     picflag=''
 else:
     vlclib=os.path.join( top_builddir, 'src', 'libvlc_pic.a' )
@@ -69,7 +69,7 @@
 	ldflags = []
 	if os.sys.platform == 'darwin':
 	    ldflags = "-read_only_relocs warning".split()
-        ldflags.extend(os.popen('%s --libs vlc %s builtin' % (vlcconfig,
+	ldflags.extend(os.popen('%s --libs vlc %s' % (vlcconfig,
 							      picflag), 
 				'r').readline().rstrip().split())
 	if os.sys.platform == 'darwin':
--- origsrc/vlc-0.8.6i/configure.ac	2008-07-17 15:20:28.859375000 -0500
+++ src/vlc-0.8.6i/configure.ac	2008-07-17 15:21:06.406250000 -0500
@@ -611,7 +611,7 @@
 
 # Win32 style
 if test "${ac_cv_have_plugins}" = "no"; then
-  if test "${SYS}" = "mingw32" -o "${SYS}" = "cygwin"; then
+  if test "${SYS}" = "mingw32"; then
     AC_CHECK_LIB(kernel32, main,
      [VLC_ADD_LDFLAGS([vlc],[-lkernel32])
       AC_DEFINE(HAVE_DL_WINDOWS, 1, [Define if you have Windows' LoadLibrary])
@@ -1860,7 +1860,7 @@
   [  --enable-dshow          Win32 DirectShow support (default enabled on Win32)])
 if test "${enable_dshow}" != "no"
 then
-  if test "${SYS}" = "mingw32" -o "${SYS}" = "cygwin"
+  if test "${SYS}" = "mingw32"
   then
     AC_LANG_PUSH(C++)
       AC_CHECK_HEADERS(dshow.h,
@@ -3350,7 +3350,7 @@
 then
   AC_CHECK_HEADERS(theora/theora.h, [
     AC_CHECK_LIB(theora, theora_granule_time, [
-      if test "${SYS}" = "mingw32"; then
+      if test "${SYS}" = "mingw32" -o "${SYS}" = "cygwin"; then
         VLC_ADD_PLUGINS([theora])
       else
         VLC_ADD_BUILTINS([theora])
@@ -3372,7 +3372,7 @@
   PKG_CHECK_MODULES(DIRAC,dirac, [
       VLC_ADD_PLUGINS([dirac])
       VLC_ADD_CFLAGS([dirac],[$DIRAC_CFLAGS])
-      VLC_ADD_LDFLAGS([dirac],[$DIRAC_LIBS -lstdc++]) ],[
+      VLC_ADD_LDFLAGS([dirac],[$DIRAC_LIBS]) ],[
       AC_MSG_ERROR([libdirac doesn't appear to be installed on you system.])
   ])
 fi
@@ -3944,7 +3944,7 @@
   [  --enable-directx        Win32 DirectX support (default enabled on Win32)])
 if test "${enable_directx}" != "no"
 then
-  if test "${SYS}" = "mingw32" -o "${SYS}" = "mingwce" -o "${SYS}" = "cygwin"
+  if test "${SYS}" = "mingw32" -o "${SYS}" = "mingwce"
   then
     AC_ARG_WITH(directx,
     [    --with-directx=PATH   Win32 DirectX headers])
@@ -4146,7 +4146,7 @@
 AC_ARG_ENABLE(wingdi,
   [  --enable-wingdi         Win32 GDI module (default enabled on Win32)])
 if test "${enable_wingdi}" != "no"; then
-  if test "${SYS}" = "mingw32" -o "${SYS}" = "cygwin"; then
+  if test "${SYS}" = "mingw32"; then
     VLC_ADD_PLUGINS([wingdi])
     VLC_ADD_LDFLAGS([wingdi],[-lgdi32])
   fi
@@ -4438,7 +4438,7 @@
     fi
   fi
 
-  if test "${skins2_missing_lib}" = "no" && (test "${SYS}" = "mingw32" -o "${SYS}" = "cygwin"); then
+  if test "${skins2_missing_lib}" = "no" && test "${SYS}" = "mingw32"; then
     VLC_ADD_PLUGINS([skins2])
     ALIASES="${ALIASES} svlc"
     VLC_ADD_CPPFLAGS([skins2],[-U_OFF_T_ -U_off_t -Imodules/gui/skins2 -DWIN32_SKINS])
@@ -5099,7 +5099,7 @@
   [    --with-wine-sdk-path=PATH path to wine sdk])
 if test "${enable_activex}" != "no"
 then
-  if test "${SYS}" = "mingw32" -o "${SYS}" = "cygwin"
+  if test "${SYS}" = "mingw32"
   then
     AC_CHECK_PROGS(MIDL, [midl], no)
     if test "${with_wine_sdk_path}" != ""
@@ -5355,7 +5355,7 @@
 dnl
 dnl Microsoft Layer for Unicode for WIN9x/ME
 dnl
-if test "${SYS}" = "mingw32" -o "${SYS}" = "cygwin"
+if test "${SYS}" = "mingw32"
 then
     AC_CHECK_LIB(unicows, LoadUnicowsSymbol, [
         VLC_ADD_LDFLAGS([vlc],[-Wl,--exclude-libs,libunicows.a -lunicows])
--- origsrc/vlc-0.8.6i/doc/Makefile.am	2008-07-08 15:59:22.000000000 -0500
+++ src/vlc-0.8.6i/doc/Makefile.am	2008-07-17 15:21:06.421875000 -0500
@@ -1,7 +1,5 @@
 NULL =
 
-docdir = $(datadir)/doc/@PACKAGE@
-
 doc_DATA = \
 	bugreport-howto.txt \
 	fortunes.txt \
--- origsrc/vlc-0.8.6i/modules/codec/faad.c	2008-07-08 15:59:23.000000000 -0500
+++ src/vlc-0.8.6i/modules/codec/faad.c	2008-07-17 15:21:06.421875000 -0500
@@ -359,6 +359,8 @@
         p_dec->fmt_out.audio.i_rate = frame.samplerate;
         p_dec->fmt_out.audio.i_channels = frame.channels;
 
+
+#ifdef NEAACDECAPI
         /* Adjust stream info when dealing with SBR/PS */
         if( (p_sys->b_sbr != frame.sbr || p_sys->b_ps != frame.ps) &&
             p_dec->p_parent->i_object_type == VLC_OBJECT_INPUT )
@@ -383,6 +385,7 @@
             }
             p_sys->b_sbr = frame.sbr; p_sys->b_ps = frame.ps;
         }
+#endif
 
         /* Convert frame.channel_position to our own channel values */
         p_dec->fmt_out.audio.i_physical_channels = 0;
--- origsrc/vlc-0.8.6i/modules/codec/realaudio.c	2008-07-08 15:59:23.000000000 -0500
+++ src/vlc-0.8.6i/modules/codec/realaudio.c	2008-07-17 15:21:06.437500000 -0500
@@ -460,6 +460,9 @@
 
 static int OpenWin32Dll( decoder_t *p_dec, char *psz_path, char *psz_dll )
 {
+#if defined(__CYGWIN__)
+    return OpenNativeDll( p_dec, psz_path, psz_dll );
+#endif
 #if defined(LOADER) || defined(WIN32)
     decoder_sys_t *p_sys = p_dec->p_sys;
     void *handle = 0, *context = 0;
--- origsrc/vlc-0.8.6i/modules/gui/ncurses.c	2008-07-08 15:59:23.000000000 -0500
+++ src/vlc-0.8.6i/modules/gui/ncurses.c	2008-07-17 15:21:06.453125000 -0500
@@ -99,7 +99,7 @@
 vlc_module_begin();
     set_shortname( "Ncurses" );
     set_description( _("Ncurses interface") );
-    set_capability( "interface", 10 );
+    set_capability( "interface", 40 );
     set_category( CAT_INTERFACE );
     set_subcategory( SUBCAT_INTERFACE_MAIN );
     set_callbacks( Open, Close );
--- origsrc/vlc-0.8.6i/modules/gui/skins2/src/skin_common.hpp	2008-07-08 15:59:23.000000000 -0500
+++ src/vlc-0.8.6i/modules/gui/skins2/src/skin_common.hpp	2008-07-17 15:21:06.468750000 -0500
@@ -76,6 +76,7 @@
     return res;
 }
 
+#ifndef __CYGWIN__
 /// Wrapper around FromWide, to avoid the need to call free()
 static inline string sFromWide( const wstring &rWide )
 {
@@ -84,6 +85,7 @@
     free( s );
     return res;
 }
+#endif
 
 /// Wrapper around ToLocale, to avoid the need to call LocaleFree()
 static inline string sToLocale( const string &rUTF8 )
--- origsrc/vlc-0.8.6i/modules/misc/freetype.c	2008-07-08 15:59:24.000000000 -0500
+++ src/vlc-0.8.6i/modules/misc/freetype.c	2008-07-17 15:21:06.468750000 -0500
@@ -55,7 +55,7 @@
 #elif defined( WIN32 )
 #define DEFAULT_FONT "" /* Default font found at run-time */
 #else
-#define DEFAULT_FONT "/usr/share/fonts/truetype/freefont/FreeSerifBold.ttf"
+#define DEFAULT_FONT "/usr/share/fonts/TTF/DejaVuSerif-Bold.ttf"
 #endif
 
 #if defined(HAVE_FRIBIDI)
--- origsrc/vlc-0.8.6i/modules/misc/network/ipv4.c	2008-07-08 15:59:24.000000000 -0500
+++ src/vlc-0.8.6i/modules/misc/network/ipv4.c	2008-07-17 15:21:06.484375000 -0500
@@ -265,7 +265,7 @@
         if (( rem.sin_addr.s_addr != INADDR_ANY )
          /*&& ((ntohl (loc.sin_addr.s_addr) >> 24) == 232)*/)
         {
-#ifndef IP_ADD_SOURCE_MEMBERSHIP
+#if !defined( IP_ADD_SOURCE_MEMBERSHIP ) || defined ( __CYGWIN__ )
             errno = ENOSYS;
 #else
             struct ip_mreq_source imr;
--- origsrc/vlc-0.8.6i/src/Makefile.am	2008-07-08 15:59:25.000000000 -0500
+++ src/vlc-0.8.6i/src/Makefile.am	2008-07-17 15:21:06.500000000 -0500
@@ -122,16 +122,16 @@
 
 stamp-builtins: Makefile ../vlc-config ../config.status
 if USE_LIBTOOL
-	@for c in `$(VLC_CONFIG) --libs builtin`; do \
+	@for c in `$(VLC_CONFIG) --target builtin`; do \
 		case $$c in \
-        		../modules/*.a) echo $$c ;; \
+        		../modules/*.la) echo $$c ;; \
 		esac ; \
 	done | \
-	sed -e 's,^\(.*\)/\([^/]*\)\.a$$,cd \1 \&\& $(MAKE) \2_builtin.la,g' | \
+	sed -e 's,^\(.*\)/\([^/]*\)$$,cd \1 \&\& $(MAKE) \2,g' | \
 	while read cmd; do echo $$cmd; eval "($$cmd)" || exit $$? ; done
 endif
 if BUILD_SHARED
-	@for c in `$(VLC_CONFIG) --libs builtin`; do \
+	@for c in `$(VLC_CONFIG) --target builtin`; do \
 		case $$c in \
 			../modules/*.a) echo $$c ;; \
 		esac ; \
@@ -156,7 +156,7 @@
 libvlc_la_CXXFLAGS = `$(VLC_CONFIG) --cxxflags vlc`
 libvlc_la_OBJCFLAGS = `$(VLC_CONFIG) --objcflags vlc`
 libvlc_la_LDFLAGS = \
-	`$(VLC_CONFIG) --libs vlc builtin|sed -e 's/\(modules\/[^ ]*\)\.a /\1_builtin.la /g'` \
+	`$(VLC_CONFIG) --libs vlc builtin` \
 	-version-info 0:8:0 \
 	-no-undefined
 libvlc_la_DEPENDENCIES = stamp-builtins
--- origsrc/vlc-0.8.6i/src/stream_output/sap.c	2008-07-08 16:08:36.000000000 -0500
+++ src/vlc-0.8.6i/src/stream_output/sap.c	2008-07-17 15:21:06.531250000 -0500
@@ -279,7 +279,7 @@
 
     switch( addr.ss_family )
     {
-#if defined (HAVE_INET_PTON) || defined (WIN32)
+#if defined (AF_INET6) || defined (WIN32)
         case AF_INET6:
         {
             /* See RFC3513 for list of valid IPv6 scopes */
@@ -456,7 +456,7 @@
     psz_head[2] = (i_hash & 0xFF00) >> 8; /* Msg id hash */
     psz_head[3] = (i_hash & 0xFF);        /* Msg id hash 2 */
 
-#if defined (HAVE_INET_PTON) || defined (WIN32)
+#if defined (AF_INET6) || defined (WIN32)
     if( b_ipv6 )
     {
         inet_pton( AF_INET6, /* can't fail */
--- origsrc/vlc-0.8.6i/vlc-config.in.in	2008-07-08 15:59:25.000000000 -0500
+++ src/vlc-0.8.6i/vlc-config.in.in	2008-07-17 15:21:06.546875000 -0500
@@ -267,7 +267,8 @@
     for module in `echo "${builtins}"`; do
       register_targets "${module}"
     done
-    for target in `echo "${list}"`; do printf "${top_builddir}modules/${target}.a "; done
+@USE_LIBTOOL_TRUE@    for target in `echo "${list}"`; do printf "${top_builddir}modules/${target}_builtin.la "; done
+@USE_LIBTOOL_FALSE@    for target in `echo "${list}"`; do printf "${top_builddir}modules/${target}.a "; done
     printf '\n'
   fi
   exit 0
@@ -320,11 +321,13 @@
       register_targets "${module}"
       register_flags "${module}"
     done
-    for target in `echo "${list}"`; do printf "${top_builddir}modules/${target}.a "; done
+@USE_LIBTOOL_TRUE@    for target in `echo "${list}"`; do printf "${top_builddir}modules/${target}_builtin.la "; done
+@USE_LIBTOOL_FALSE@    for target in `echo "${list}"`; do printf "${top_builddir}modules/${target}.a "; done
   fi
   if test "${echo_external}" = yes; then
     for module in `echo "${builtins}"`; do
-      ldflags="${ldflags} @libdir@/vlc/lib${module}.a"
+@USE_LIBTOOL_TRUE@       :
+@USE_LIBTOOL_FALSE@      ldflags="${ldflags} @libdir@/vlc/lib${module}.a"
     done
     for module in `echo "${builtins}"`; do
       register_flags "${module}"
